### PR TITLE
Make `RID_Owner` lock-free for fetching.

### PR DESCRIPTION
**Pull request by DarioSamo**
Taking over https://github.com/godotengine/godot/pull/86333 from @reduz to fix some of the newest conflicts.

### CODE REVIEW

**Review by reduz — APPROVED** [review]

File: core/templates/rid_owner.h [hunk 1]
```diff
@@ -419,8 +416,8 @@ class RID_PtrOwner {
 		alloc.set_description(p_descrption);
 	}
 
-	RID_PtrOwner(uint32_t p_target_chunk_byte_size = 65536) :
-			alloc(p_target_chunk_byte_size) {}
+	RID_PtrOwner(uint32_t p_target_chunk_byte_size = 65536, uint32_t p_maximum_amount_of_elements = 262144) :
```
Comment (AThousandShips) [inline core/templates/rid_owner.h:419]:
```suggestion
	RID_PtrOwner(uint32_t p_target_chunk_byte_size = 65536, uint32_t p_maximum_number_of_elements = 262144) :
```
"Number" for countable things

File: core/templates/rid_owner.h [hunk 2]
```diff
@@ -473,8 +470,8 @@ class RID_Owner {
 	void set_description(const char *p_descrption) {
 		alloc.set_description(p_descrption);
 	}
-	RID_Owner(uint32_t p_target_chunk_byte_size = 65536) :
-			alloc(p_target_chunk_byte_size) {}
+	RID_Owner(uint32_t p_target_chunk_byte_size = 65536, uint32_t p_maximum_amount_of_elements = 262144) :
```
Comment (AThousandShips) [inline core/templates/rid_owner.h:473]:
```suggestion
	RID_Owner(uint32_t p_target_chunk_byte_size = 65536, uint32_t p_maximum_number_of_elements = 262144) :
```

File: core/templates/rid_owner.h [hunk 3]
```diff
@@ -294,43 +287,49 @@ class RID_Alloc : public RID_AllocBase {
...
+			mutex.unlock();
 		}
 	}
 
 	void set_description(const char *p_descrption) {
 		description = p_descrption;
 	}
 
-	RID_Alloc(uint32_t p_target_chunk_byte_size = 65536) {
+	RID_Alloc(uint32_t p_target_chunk_byte_size = 65536, uint32_t p_maximum_amount_of_elements = 262144) {
```
Comment (AThousandShips) [inline core/templates/rid_owner.h:326]:
```suggestion
	RID_Alloc(uint32_t p_target_chunk_byte_size = 65536, uint32_t p_maximum_number_of_elements = 262144) {
```

File: core/templates/rid_owner.h (hunk 1 above)
Comment (DarioSamo) [reply core/templates/rid_owner.h]: Done.

**Review by RandomShaper — APPROVED** [review]
As far as I can tell, some extra memory ordering enforcement is needed here. That said, I haven't been able to make this misbehave on ARM, the weakest ordering arch we're targeting, in the experiments I'm doing in #97654. I may make a future PR adding those even if it's only for theoretical correctness or future-proofing.

For now, I can't but greenlight this.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**